Indicators/RelativeDailyVolume.cs

```diff
@@ -0,0 +1,141 @@
+    /// 
+    /// Current volume from open to current time of day / Average over the past x days from open to current time of day
+    /// </summary>
+    public class RelativeDailyVolume : TradeBarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private readonly List<decimal> _currentVolume = new List<decimal>();
+        private readonly Dictionary<DateTime, decimal> _historicalVolumes = new Dictionary<DateTime, decimal>();
+        private readonly RollingWindow<DateTime> _rollingTime;
+        private readonly RollingWindow<decimal> _rollingData;
+        private int _previousDay;
+
+        /// <summary>
+        /// Gets a flag indicating when the indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples >= WarmUpPeriod;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Initializeds a new instance of the RelativeVolume class using the specified period
+        /// </summary>
+        /// <param name="period">The period over which to perform to computation</param>
+        public RelativeDailyVolume(int period = 14)
+            : this($"RDV({period})", period)
+        {
+        }
+        /// <summary>
+        /// Creates a new RelativeVolume indicator with the specified period
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of this indicator</param>
+        public RelativeDailyVolume(string name, int period)
+            : base(name)
+        {
+            _previousDay = -1;
+            WarmUpPeriod = period;
+            _rollingTime = new RollingWindow<DateTime>(period * 2);
+            _rollingData = new RollingWindow<decimal>(period * 2);
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// </summary>
+        /// <param name="input">The input value to this indicator on this time step</param>
+        /// <returns>A a value for this indicator</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            if (input.Time.Day != _previousDay)
+            {
+                _currentVolume.Clear();
+                _previousDay = input.Time.Day;
+            }
+
+            _currentVolume.Add(input.Volume);
+            _historicalVolumes.Add(input.Time, input.Volume);
+            _rollingTime.Add(input.Time);
+            _rollingData.Add(input.Volume);
+
+            if (!IsReady) { return 0; }
+
+            var todaysTotal = _currentVolume.Sum();
+            var timeList = _rollingTime.ToList();
+            var dataList = _rollingData.ToList();
+            int listCount = timeList.Count;
+            var currentDay = -1;
+            int listDifference = (listCount - WarmUpPeriod) - 1;
```

Comment:
`int` -> can just use var in these cases

> if (listDifference != -1) { timeList.Skip(listDifference); }

In Lean code style usually we place each curly Braces in a new line